elasticsearch/docker/build/elasticsearch/bin/docker-entrypoint.sh

```diff
@@ -7,7 +7,7 @@ umask 0002
 run_as_other_user_if_needed() {
     if [[ "$(id -u)" == "0" ]]; then
         # If running as root, drop to specified UID and run command
-        exec chroot --userspec=1000 / "${@}"
+        exec chroot --userspec=${PUID:-${DEFAULT_UID}} / "${@}"
```

Comment:
`run_as_other_user_if_needed` will run the specified process as `PUID` (defaulting to `DEFAULT_UID`, which is specified as 1000 in the Dockerfile) rather than just hard-coding 1000.